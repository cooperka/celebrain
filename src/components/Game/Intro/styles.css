.Intro .start {
  margin-top: 1rem;
}
