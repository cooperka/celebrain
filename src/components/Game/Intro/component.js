// @flow

import * as R from 'ramda';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  withStyles,
  Button, Switch, Checkbox, TextField,
  FormLabel, FormControl, FormGroup, FormControlLabel,
} from 'material-ui';

import { setNumCelebs, setOnlyFirstNames, setPopularity, setGameState } from '../actions';
import { gameState } from '../reducers';

import './styles.css';

type Props = {
  numCelebs: number,
  onlyFirstNames: boolean,
  popularityGroups: Array<boolean>,

  handleSetNumCelebs: (number) => void,
  handleSetOnlyFirstNames: (boolean) => void,
  handleSetPopularity: (boolean, number) => void,
  handleStartGame: () => void,
};

class Intro extends Component<Props> {

  static mapStateToProps(state) {
    return {
      numCelebs: state.game.numCelebs,
      onlyFirstNames: state.game.onlyFirstNames,
      popularityGroups: state.game.popularityGroups,
    };
  }

  static mapDispatchToProps(dispatch) {
    return {
      handleSetNumCelebs: (numCelebs) => {
        dispatch(setNumCelebs(numCelebs));
      },
      handleSetOnlyFirstNames: (isChecked) => {
        dispatch(setOnlyFirstNames(isChecked));
      },
      handleSetPopularity: (isChecked, groupIndex) => {
        dispatch(setPopularity(isChecked, groupIndex));
      },
      handleStartGame: () => {
        dispatch(setGameState(gameState.MEMORIZE));
      },
    };
  }

  renderPopularityControls() {
    const { popularityGroups, handleSetPopularity } = this.props;

    const labels = [
      '75th percentile (popular)',
      '50-75th percentile',
      '25-50th percentile',
      '0-25th percentile (uncommon)',
    ];

    return (
      <FormGroup>
        {popularityGroups.map((isChecked, index) => (
          <FormControlLabel
            // eslint-disable-next-line react/no-array-index-key
            key={index}
            control={
              <Checkbox
                checked={isChecked}
                onChange={(_, checked) => handleSetPopularity(checked, index)}
              />
            }
            label={labels[index]}
          />
        ))}
      </FormGroup>
    );
  }

  render() {
    const { numCelebs, onlyFirstNames, handleSetOnlyFirstNames, handleSetNumCelebs, handleStartGame } = this.props;

    return (
      <div className="Intro">
        <div className="title">New game</div>
        <FormControl component="fieldset">
          <FormLabel component="legend">Settings</FormLabel>
          <FormControlLabel
            control={
              <TextField
                value={numCelebs}
                onChange={(event) => handleSetNumCelebs(event.target.value)}
                type="number"
                margin="normal"
                className="num-celebs"
              />
            }
            label="Number of celebrities"
          />
          <FormControlLabel
            control={
              <Switch
                checked={onlyFirstNames}
                onChange={(_, isChecked) => handleSetOnlyFirstNames(isChecked)}
              />
            }
            label="Only learn first names"
          />
          <FormLabel component="label">Celebrity popularity</FormLabel>
          {this.renderPopularityControls()}
          <Button className="start" onClick={handleStartGame}>Start</Button>
        </FormControl>
      </div>
    );
  }

}

const styles = () => ({
});

export default R.compose(
  withStyles(styles),
  connect(Intro.mapStateToProps, Intro.mapDispatchToProps),
)(Intro);
