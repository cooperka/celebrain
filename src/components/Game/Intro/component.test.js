// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import Intro from './component';

it('renders', () => {
  const tree = renderer.create(<Intro />);
  expect(tree.toJSON()).toMatchSnapshot();
});
