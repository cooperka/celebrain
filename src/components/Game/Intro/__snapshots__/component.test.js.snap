// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Intro"
>
  <div
    className="title"
  >
    New game
  </div>
  <fieldset
    className="MuiFormControl-root-1"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <legend
      className="MuiFormLabel-root-5"
    >
      Settings
    </legend>
    <label
      className="MuiFormControlLabel-root-9"
    >
      <span
        className="MuiSwitch-root-12"
      >
        <span
          className="MuiButtonBase-root-32 MuiIconButton-root-26 MuiSwitchBase-root-21 MuiSwitchBase-default-23 MuiSwitch-default-16"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-31"
          >
            <span
              className="MuiSwitch-icon-14"
            />
            <input
              checked={undefined}
              className="MuiSwitchBase-input-22"
              disabled={false}
              id={undefined}
              name={undefined}
              onChange={[Function]}
              tabIndex={undefined}
              type="checkbox"
              value={undefined}
            />
          </span>
          <span
            className="MuiTouchRipple-root-34"
          />
        </span>
        <span
          className="MuiSwitch-bar-13"
        />
      </span>
      <span
        className="MuiTypography-root-41 MuiTypography-body1-50 MuiFormControlLabel-label-11"
      >
        Only learn first names
      </span>
    </label>
    <label
      className="MuiFormLabel-root-5"
    >
      Celebrity popularity
    </label>
    <div
      className="MuiFormGroup-root-65"
    >
      <label
        className="MuiFormControlLabel-root-9"
      >
        <span
          className="MuiButtonBase-root-32 MuiIconButton-root-26 MuiSwitchBase-root-21 MuiSwitchBase-default-23 MuiCheckbox-default-67 MuiSwitchBase-checked-24 MuiCheckbox-checked-68 MuiCheckbox-checkedSecondary-70"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role={undefined}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label-31"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-72 MuiSvgIcon-fontSize-78"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <input
              checked={true}
              className="MuiSwitchBase-input-22"
              disabled={false}
              id={undefined}
              name={undefined}
              onChange={[Function]}
              tabIndex={undefined}
              type="checkbox"
              value={undefined}
            />
          </span>
          <span
            className="MuiTouchRipple-root-34"
          />
        </span>
        <span
          className="MuiTypography-root-41 MuiTypography-body1-50 MuiFormControlLabel-label-11"
        >
          75th percentile (popular)
        </span>
      </label>
    </div>
    <button
      className="MuiButtonBase-root-32 MuiButton-root-79 start"
      disabled={false}
      onBlur={[Function]}
      onClick={undefined}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label-80"
      >
        Start
      </span>
      <span
        className="MuiTouchRipple-root-34"
      />
    </button>
  </fieldset>
</div>
`;
